src/components/PostsList/PostsList.tsx

```diff
@@ -1,37 +1,56 @@
 import React from 'react';
+import { Loader } from '../Loader';
 import './PostsList.scss';
 
-export const PostsList: React.FC = () => (
-  <div className="PostsList">
-    <h2>Posts:</h2>
+type Props = {
+  posts: Post[],
+  isLoading: boolean,
+  selectedPostId: number | null,
+  onPostSelect: CallableFunction,
+};
 
-    <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Close
-        </button>
-      </li>
+export const PostsList: React.FC<Props> = ({
+  posts,
+  isLoading,
+  selectedPostId,
+  onPostSelect,
+}) => {
+  return (
+    <div className="PostsList">
+      <h2>Posts:</h2>
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <ul className="PostsList__list" data-cy="postDetails">
+          {posts.map(post => (
+            <li key={post.id} className="PostsList__item">
+              <div>
+                <b>{`[User #${post.userId}]: `}</b>
+                {post.title}
+              </div>
+              {selectedPostId === post.id
+                ? (
+                  <button
+                    type="button"
+                    className="PostsList__button button"
+                    onClick={() => onPostSelect(0)}
+                  >
+                    Close
+                  </button>
+                ) : (
+                  <button
+                    type="button"
+                    className="PostsList__button button"
+                    onClick={() => onPostSelect(post.id)}
+                  >
+                    Open
+                  </button>
```

Comment:
same button but different logic for callback - so check it in callback

Reply:
done